### PR TITLE
add wrap_func, stop, reset, metrics and minor changes

### PR DESCRIPTION
**Pull request by zlobspb**
add graphite.wrap_func which adds avg, min, max, rpm stat to function with one line

add functions:
graphite.stop
graphite.reset
graphite.metrics

local added for nt variable

forgotten logging removed

### CODE REVIEW

File: graphite.lua [hunk 1]
```diff
@@ -219,50 +221,82 @@ local function collect_stats()
...
+		stat_fiber = nil
+	end
+
+	if sock ~= nil then
+		sock:close()
+		sock = nil
+	end
+end
+
+_M.reset = function()
```
Comment (viciious) [inline graphite.lua]: The semantics (the fact that stop() is also called) of this function is unclear from its naming. I'd rather prefer to have an optional argument passed to stop() instead.

File: graphite.lua [hunk 2]
```diff
@@ -404,4 +437,43 @@ _M.inc = function(name)
...
+
+		status['host'] = host
+		status['port'] = port
+		status['prefix'] = prefix
+	end
+
+	return status
+end
+
+local function stat_tail(stat_name, start_time, ...)
```
Comment (viciious) [inline graphite.lua]: Why "stat_tail"?..

File: graphite.lua [hunk 3]
```diff
@@ -404,4 +437,43 @@ _M.inc = function(name)
...
+	local delta = fiber.time64() - start_time
+	_M.avg_per_min(stat_name['avg'], delta)
+	_M.min_per_min(stat_name['min'], delta)
+	_M.max_per_min(stat_name['max'], delta)
+	_M.sum_per_min(stat_name['rpm'], 1)
+	return ...
+end
+
+_M.wrap_func = function(name, func)
```
Comment (viciious) [inline graphite.lua]: Again, the semantics of this function is unclear from its naming.

File: README.md [hunk 4]
```diff
@@ -50,8 +50,21 @@ graphite.send(name, res, timestamp)
...
-`graphite.add_sec_metric('delete_rps_max', function() return box.stat().DELETE.rps end, graphite.max)`
+``` lua
+graphite.add_sec_metric('delete_rps_max', function() return box.stat().DELETE.rps end, graphite.max)
+
+graphite.add('requests', 1)
+
+graphite.max_per_min('max_mysql_query_time', query_time)
 
-`graphite.add('requests', 1)`
+-- graphite.wrap_func adds for function dostuff following stats with one line:
```
Comment (viciious) [inline README.md]: Please rephrase this to something like "An utility function, handy for adding the following set of metrics with a single line of code".
Why this particular set anyway?  How many custom rpm metrics do you have?

**Review by viciious — CHANGES_REQUESTED** [review]

File: graphite.lua (hunk 1 above)
Comment (zlobspb) [reply graphite.lua]: will do it

File: graphite.lua (hunk 2 above)
Comment (zlobspb) [reply graphite.lua]: wrap_func_with_stat_tail - ok?

File: graphite.lua (hunk 3 above)
Comment (zlobspb) [reply graphite.lua]: maybe wrap_func_with_stat? can you suggest enything?

File: README.md (hunk 4 above)
Comment (zlobspb) [reply README.md]: It's just I'm using this set. Do you want to add anything?

### Utility functions

``` lua
-- handy wrapper to add func_name_avg, func_name_min, func_name_max, func_name_rpm metrics to your function with a single line of code
graphite.wrap_func_with_stat(name, func)
```

## Usage examples

``` lua
local function dostuff()
end

dostuff = graphite.wrap_func_with_stat('dostuff', dostuff)
```
